checker/index.php

```diff
@@ -127,8 +129,8 @@
 if ($_POST["validate_uri"])
 {
 	$_POST['uri'] = htmlentities($_POST['uri']);
-	
-	$uri = Utility::getValidURI($addslashes($_POST["uri"]));
+	$url=$DAO->addSlash($_POST["uri"]);
```

Comment:
However, this is not how DAO class should be used. `DAO.class.php` is used as a base class by all higher level DAO classes for database manipulation. It doesn't make sense for it to deal with non-db related scenarios.

The use of `addslahes` here is a bug from beginning. If you take a look on [the implementation of getValidURI()](https://github.com/inclusive-design/AChecker/blob/master/include/classes/Utility.class.php#L50), the method doesn't require `uri` as an input of a SQL statement, which means it doesn't make sense for this variable to be sanitized with `addslashes`, but does make sense to sanitize it [PHP built-in method filter_var()](http://us.php.net/manual/en/function.filter-var.php). 

Although [this discussion of methods for sanitizing user input with PHP](https://stackoverflow.com/questions/129677/whats-the-best-method-for-sanitizing-user-input-with-php) is a bit old, it still quite applicable and worth a read.